CMakeLists.txt

```diff
@@ -710,341 +710,26 @@ endif()
 # ---------------------------------------------------------------
 
 #-------------------------------- Source files -------------------------------
-set( SRC_MAIN "src/main.cpp" )
-file( GLOB_RECURSE SRC_APP		"src/Application/*.cpp" )
-file( GLOB_RECURSE SRC_CORE		"src/Core/*.cpp" )
-file( GLOB_RECURSE SRC_DATA		"src/Data/*.cpp" )
-file( GLOB_RECURSE SRC_IMPORTER	"src/Importer/*.cpp" )
-file( GLOB_RECURSE SRC_LAYOUT	"src/Layout/*.cpp" )
-file( GLOB_RECURSE SRC_LUAGRAPH  "src/LuaGraph/*.cpp")
-file( GLOB_RECURSE SRC_LUAINTERFACE  "src/LuaInterface/*.cpp")
-file( GLOB_RECURSE SRC_LUATYPES  "src/LuaTypes/*.cpp")
-file( GLOB_RECURSE SRC_MANAGER	"src/Manager/*.cpp" )
-file( GLOB_RECURSE SRC_MATH		"src/Math/*.cpp" )
-file( GLOB_RECURSE SRC_MODEL	"src/Model/*.cpp" )
-file( GLOB_RECURSE SRC_NETWORK	"src/Network/*.cpp" )
-file( GLOB_RECURSE SRC_OSGQTBROWSER	"src/OsgQtBrowser/*.cpp" )
-file( GLOB_RECURSE SRC_QOSG		"src/QOSG/*.cpp" )
-file( GLOB_RECURSE SRC_UTIL		"src/Util/*.cpp" )
-file( GLOB_RECURSE SRC_VIEWER	"src/Viewer/*.cpp" )
-file( GLOB_RECURSE SRC_CLUSTERING	"src/Clustering/*.cpp")
-file( GLOB SRC_REPOSITORY		"src/Repository/Git/*.cpp" )
-file( GLOB_RECURSE SRC_GITLIB   "src/Repository/Git/GitLib/*.cpp" )
-file( GLOB_RECURSE SRC_SPECIALMATRIX   "src/SpecialMatrix/*.cpp" )
-file( GLOB_RECURSE SRC_CITY		"src/City/*.cpp" )
-file( GLOB_RECURSE SRC_SHAPES	"src/Shapes/*.cpp" )
-
-# OpenCV
-file( GLOB_RECURSE SRC_OPENCV	"src/OpenCV/*.cpp" )
-file( GLOB_RECURSE SRC_QOPENCV	"src/QOpenCV/*.cpp" )
-
-# Aruco
-file( GLOB_RECURSE SRC_ARUCO	"src/Aruco/*.cpp" )
-
-# Kinect
-file( GLOB_RECURSE SRC_KINECT	"src/Kinect/*.cpp" )
-
-# SpeechSDK
-file( GLOB_RECURSE SRC_SPEECH	"src/Speech/*.cpp" )
+# add cmake modul that generates a list of source files
+include( cmake/Sources.cmake )
```

Comment:
toto nie je celkom dobre riesenie:

- presunut cmake/Sources.cmake do src/Sources.cmake
- resp. by sa to malo spravit cez add_subdirectory